### PR TITLE
update CHERI LLVMs

### PR DESCRIPTION
**Pull request by brooksdavis**
Sync the assorted CHERI LLVM ports with upstream.

### CODE REVIEW

File: devel/llvm-cheri/Makefile [hunk 1]
```diff
@@ -52,14 +52,13 @@ SUB_LIST+=	LLVM_PREFIX="${LLVM_PREFIX}" LLVM_SUFFIX="${LLVM_SUFFIX}"
...
 CMAKE_ARGS+=	-DLLVM_BUILD_LLVM_DYLIB=ON -DLLVM_LINK_LLVM_DYLIB=ON
-CMAKE_ARGS+=	-DCMAKE_INSTALL_MANDIR:PATH="share/man"
 CMAKE_ARGS+=	-DLLVM_PARALLEL_LINK_JOBS=1
 CMAKE_ARGS+=	-DLLVM_ENABLE_Z3_SOLVER=OFF
 CMAKE_ARGS+=	-DLLVM_ENABLE_LIBXML2=OFF
 CMAKE_ARGS+=	-DLLVM_ENABLE_LIBEDIT=OFF
 CMAKE_ARGS+=	-DLLVM_TARGETS_TO_BUILD="${LLVM_TARGETS}"
 
-.ifndef MASTERDIR
+.ifndef USE_GITLAB
```
Comment (eupharina) [inline devel/llvm-cheri/Makefile:61]: I rebased #109 onto this branch. llvm-morello-csa also uses github, and this code overrides it's github settings. Maybe `GH_*` variables for llvm-cheriot could be defined in `llvm-cheriot/Makefile` instead?

Comment (brooksdavis) [reply devel/llvm-cheri/Makefile:61]: Indeed, this worked a bit by accident (you can retrieve a git hash present in any fork for the main repo or any other fork.). I think the answer is to make these ?= assignments.  I'll do that.

**Review by kwitaszczyk — APPROVED** [review]
I'm fine with merging it as it is now.

@eupharina Any objections with regards to #109?

**Comment by eupharina** [comment]
LGTM

**Pull request merged**